esmvalcore/preprocessor/_time.py

```diff
@@ -130,6 +232,22 @@ def clip_start_end_year(cube, start_year, end_year):
     ValueError
         Time ranges are outside the cube's time limits.
     """
+    if timerange:
+        start_date = timerange.split('/')[0]
+        start_date = _parse_start_date(start_date)
+
+        end_date = timerange.split('/')[1]
+        end_date = _parse_end_date(end_date)
+
+        if isinstance(start_date, isodate.duration.Duration):
+            start_date = _duration_to_date(cube, start_date, end_date, sign=-1)
+            start_date += datetime.timedelta(days=1)
```

Comment:
I think I needed to set this to seconds=1, let me recheck.

Reply:
Actually this crashes for durations without date, like `PT12H`. I can either fix it here, or since most data that is being used does not have that much of resolution in time, do it in another PR.

Reply:
If this breaks processing of ERA5 hourly data, it is important that both this and the other PR are merged in the same release.